streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -71,4 +71,20 @@ public void process(K key, V value) {
     public void close() {
         processor.close();
     }
+
+    /**
+     *
+     * @return
+     */
+    public String toString() {
+        String processorNodeString = "ProcessorNode[name=" + name;
+        if (stateStores != null && !stateStores.isEmpty()) {
+            processorNodeString += "," + "StateStores[";
+            for (String store : stateStores) {
+                processorNodeString += "store=" + store + ",";
```

Comment:
Same as other case, String concatenation should not be used inside loops as it's very inefficient (a new array is allocated for each concatenation instead of doubling the internal array as needed).